.github/workflows/build_cts_json.yaml

```diff
@@ -0,0 +1,51 @@
+name: Build 'cts.json'
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build-cts:
+    runs-on: ubuntu-latest
+
+    permissions:
+      # Allow the job to push the changed file to the repository
+      # For pull requests from forks this seems to be implicitly changed to 'read', as desired, see
+      # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#how-the-permissions-are-calculated-for-a-workflow-job
+      contents: write
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup Node.js
+      id: setup-node
+      uses: actions/setup-node@v4
+      with:
+        node-version: 'lts/*'
+
+    - name: Run build
+      run: ./build.sh
+
+    # To be safe, verify that either there are no changes or only `cts.json` has changed
+    - name: Verify no unexpected changes
+      run: |
+        # Check for changes to any file other than `cts.json`,
+        # see https://stackoverflow.com/a/29374503
+        # Note that this does not detect new untracked files
+        if ! git diff --exit-code --quiet -- . ':!cts.json'; then
+          echo "Unexpected changes:"
+          git diff -- . ':!cts.json'
+          exit 1
+        fi
+
+    # Commit and push changes; has no effect if the file did not change
+    # Important: The push event will not trigger any other workflows, see
+    # https://github.com/stefanzweifel/git-auto-commit-action?tab=readme-ov-file#commits-made-by-this-action-do-not-trigger-new-workflow-runs
+    - name: Commit & push changes
+      # Only run for push events (on `main` branch); don't run for pull requests
+      if: github.event_name == 'push'
```

Comment:
Wouldn't this trigger if someone pushes to a branch?  I think we need to check branch name instead.

Reply:
I think the `branches: - main` condition at the top of this file should limit this only to `main`.

Though I can also remove the condition there (so that it is instead run for all branches) and instead add the branch name check here.

Reply:
> I think the `branches: - main` condition at the top of this file should limit this only to `main`.

Those are an "any of" set, so any activity on the PR branch would still trigger this (that's how I have mine set up).

The only way to get this to only run on `main` is to check the branch.

---

We should also add `workflow_dispatch:` so that we can manually run it.  It's just a convenience.

Reply:
> Those are an "any of" set, so any activity on the PR branch would still trigger this

But I think in that case the `github.event_name` wouldn't be `push` but `pull_request`? (more specifically `pull_request` with [activity type `synchronize`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)?)


> We should also add `workflow_dispatch:` so that we can manually run it. It's just a convenience.

I guess I can add that, but assuming this workflow works as expected then (at least for the `main` branch) there would be no need for it I assume. And the `if` condition here would then also have to allow the `workflow_dispatch`. Should I add it nonetheless?

Reply:
> more specifically `pull_request` with activity type `synchronize`?

No, I think `pull-request` only has to do with pull request actions: opening, approving, comments, etc.

Regardless, I know it works the way I have it set up.  Please just do that.

Reply:
`synchronize` is [described as](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=synchronize#pull_request):
> A pull request's head branch was updated. For example, the head branch was updated from the base branch or new commits were pushed to the head branch.

In my short testing in my fork it looks that is also what was triggered when I made changes on the pull request branch. I assume that you are right, when the pull request branch is in the same repository (and not in a fork) it would also trigger a `push` event, but probably for the pull request branch then, and not `main`.

Either way, having a second check here does not hurt and makes it less error-prone, so thanks for the suggestion. I have changed it now.

As side note: This pull request has "Allow edits" enabled, so feel free to directly make any changes to it, if that is easier.